src/modules/paymentProcessor/StreamingPaymentProcessor.sol

```diff
@@ -482,140 +490,149 @@ contract StreamingPaymentProcessor is Module, IStreamingPaymentProcessor {
         }
     }
 
-    /// @notice used to remove the payment order with id = walletId from the activeVestingWallets[client][contributor] array.
+    /// @notice used to remove the payment order with id = walletId from the activeVestingWallets[client][paymentReceiver] array.
     /// @dev This function simply removes a particular payment order from the earlier mentioned array. The implications of removing a payment order
-    ///      from this array have to be handled outside of this function, such as checking whether the contributor is still active or not, etc.
+    ///      from this array have to be handled outside of this function, such as checking whether the paymentReceiver is still active or not, etc.
     /// @param client Address of the payment client
-    /// @param contributor Address of the contributor
+    /// @param paymentReceiver Address of the paymentReceiver
     /// @param walletId Id of the payment order that needs to be removed
     function _removePaymentForSpecificWalletId(
         address client,
-        address contributor,
+        address paymentReceiver,
         uint walletId
     ) internal {
-        uint walletIdIndex = _findActiveWalletId(client, contributor, walletId);
+        uint walletIdIndex =
+            _findActiveWalletId(client, paymentReceiver, walletId);
 
         if (walletIdIndex == type(uint).max) {
             revert Module__PaymentProcessor__InactiveWallet(
                 address(client), _msgSender(), walletId
             );
         }
 
+        uint stillReleasable =
+            releasableForSpecificWalletId(client, paymentReceiver, walletId);
+        //In case there is still something to be released
+        if (stillReleasable > 0) {
```

Comment:
This checks if the amount ready to be released is more than 0, but since when removing a payment you first claim what's ready it will always be 0 in that case. What you should check and mark as amount paid is the total amount minus the already released amount (except the amount in unclaimable).

Reply:
Has been addressed

Reply:
Looks better, but you forgot to add the unclaimable amount in the calculation since you still have to reserve that amount.

Reply:
So to my understanding, and @FHieser, I would love for you to double check me here, this function is meant to "free up" all the payment that is still held in the contract after a claim has been removed.

As the full amount is represented by `..._salary`, and all that is claimable has been claimed before this function call so as you stated in your first comment, what is left is the unclaimable amount.

This unclaimable amount has already been removed from the paymentProcessor, so to "remove" any claim to this amount we flag the unclaimable amount as paid within the paymentClient.

Reply:
I don't see it removed aywhere though, and if you remove it here then the user will not be able to retry and claim it

Reply:
So the way I stated it was not fully accurate, as we indeed start to remove the entry within this function and after, not necessary before. 

The function we are talking about, `_removePaymentForSpecificWalletId()`,  only gets called within the function `_afterClaimCleanup()`. This function again only gets called when we want to:
- delete all payments related to a paymentReceiver
- nothing more remains to be vested from the specific walletId

We remove the the wallet from the `activeVestingWallets` mapping and then continue with other steps of cleaning up (removing the vesting entry) within the function `_afterClaimCleanup()`.

So the fact still is that we want to make sure that all of the unclaimable amount can't be claimed through the paymentClient which is why we flag it as `..amountPaid()`

Reply:
Ah ok, I think I see what you mean, but I think it's still a problem. The user should be able to claim the unclaimable funds even if the payment is no longer active (since that part belongs to the user already). I think you should either change the code or add a function for that so users can just retry to claim their unclaimable funds even if the payment is no longer active.

Reply:
So there is one intended case whereby the stream of tokens should be able to be canceled in which means the user should not be able to access the claim anymore.

The other case would be that there is an accounting error somewhere, in which case `remainingReleasable` has a value when it should't, signalling that the user still has something unclaimed.

I've messaged @FHieser to give some input here as well on how to handle the latter case

Reply:
So we discussed the issue during our sync and you are right, the `unclaimableAmount` which represents failed transfer would be removed as well. This is an edge case but it is not intended behaviour so we will need to adapt the code.

Due to our time constraints and the fact that there is a lot of code in this PR which we need in main for protocol audit, we decided to go ahead and merge this even though the edge case is not taken care of. I've created an issue for this and we will refactor the code this coming work to make sure that the code works as expected.

Reply:
Here is the link #445

Reply:
The issue has been created in Linear Jfyi